x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/create_field/semantic_text/use_semantic_text.ts

```diff
@@ -65,113 +74,88 @@ export function useSemanticText(props: UseSemanticTextProps) {
+      const defaultInferenceEndpointConfig: DefaultInferenceEndpointConfig = {
+        service:
+          trainedModelId === ElasticsearchModelDefaultOptions.elser ? 'elser' : 'elasticsearch',
+        taskType:
+          trainedModelId === ElasticsearchModelDefaultOptions.elser
+            ? 'sparse_embedding'
+            : 'text_embedding',
+      };
+
+      const modelConfig = customInferenceEndpointConfig
+        ? customInferenceEndpointConfig.modelConfig
+        : {
+            service: defaultInferenceEndpointConfig.service,
+            service_settings: {
+              num_allocations: 1,
+              num_threads: 1,
+              model_id: trainedModelId,
+            },
+          };
+      const taskType: InferenceTaskType =
+        customInferenceEndpointConfig?.taskType ?? defaultInferenceEndpointConfig.taskType;
       try {
-        const response: TrainedModelConfigResponse[] | undefined =
-          await ml?.mlApi?.trainedModels.getTrainedModels(modelId, {
-            include: 'definition_status',
-          });
-        return !!response?.[0]?.fully_defined;
+        await ml?.mlApi?.inferenceModels?.createInferenceEndpoint(
+          data.inferenceId,
+          taskType,
+          modelConfig
+        );
       } catch (error) {
-        if (error.body.statusCode !== 404) {
-          throw error;
-        }
+        throw error;
       }
-      return false;
     },
-    [ml?.mlApi?.trainedModels]
+    [ml?.mlApi?.inferenceModels]
   );
 
-  const createInferenceEndpoint = (
-    trainedModelId: string,
-    defaultInferenceEndpoint: boolean,
-    data: Field
+  const handleSemanticText = async (
+    data: Field,
+    customInferenceEndpointConfig?: CustomInferenceEndpointConfig
   ) => {
-    if (data.inferenceId === undefined) {
-      throw new Error(
-        i18n.translate('xpack.idxMgmt.mappingsEditor.createField.undefinedInferenceIdError', {
-          defaultMessage: 'InferenceId is undefined while creating the inference endpoint.',
-        })
-      );
-    }
-
-    if (trainedModelId && defaultInferenceEndpoint) {
-      const modelConfig = {
-        service: 'elasticsearch',
-        service_settings: {
-          num_allocations: 1,
-          num_threads: 1,
-          model_id: trainedModelId,
-        },
-      };
-
-      ml?.mlApi?.inferenceModels?.createInferenceEndpoint(
-        data.inferenceId,
-        'text_embedding',
-        modelConfig
-      );
-    }
-  };
-
-  const handleSemanticText = async (data: Field) => {
     data.inferenceId = inferenceValue;
     if (data.inferenceId === undefined) {
       return;
     }
-
     const inferenceData = inferenceToModelIdMap?.[data.inferenceId];
-
     if (!inferenceData) {
       return;
     }
 
-    const { trainedModelId, defaultInferenceEndpoint, isDeployed, isDeployable } = inferenceData;
+    const { trainedModelId } = inferenceData;
+    dispatch({ type: 'field.addSemanticText', value: data });
+    showSuccessToasts();
```

Comment:
nitpick: we should probably only show this message if we actually start an inference model, not if it already exist (so move this down)

Reply:
If we are  showing success toasts only if we create inference endpoint, I think we need to reevaluate if we need to show successToasts [here](https://github.com/elastic/kibana/blob/04f1cedefbe67bb9ac6aa62d6b40adab20a7c73f/x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/field_parameters/select_inference_id.tsx#L169) as well.  I added this as there is a requirement in ticket description to close the flyout and show success toasts. But with this, we will be showing success toasts twice. 

I think we can close flyout and show toasts only when creating inference endpoint which will be after `Add field `button is clicked. WDYT?

Reply:
Yup, I think that makes sense!